Dockerfile

```diff
@@ -15,11 +15,18 @@ ENV SIMPLE_MODE=$SIMPLE_MODE
 RUN npm run build -- --mode production
 
 # Production stage
-FROM nginx:alpine
+FROM nginxinc/nginx-unprivileged:1.29-alpine
 
+# Copy files as root first, then change ownership
 COPY --from=builder /app/dist /usr/share/nginx/html
 COPY nginx.conf /etc/nginx/nginx.conf
 
-EXPOSE 80
+# Change ownership while still root, then switch to nginx user
+USER root
+RUN mkdir -p /etc/nginx/tmp && \
+    chown -R nginx:nginx /usr/share/nginx/html /etc/nginx/tmp
```

Comment:
Copy has chown and chmod options you can use.

https://docs.docker.com/reference/dockerfile#copy---chown---chmod